### PR TITLE
feat: Added staking overview page

### PR DESCRIPTION
**Pull request by Teja2045**
feat : add staking overview page

### CODE REVIEW

File: frontend/src/pages/stakingOverview/Validators.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          <TableHead>
+            <TableRow >
+              <TableCell>Validator Name</TableCell>
+              <TableCell>Staked Amount</TableCell>
+              <TableCell>Rewards</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {props.validators.map((validator) => (
+              <Validator validator={validator} key={validator.validatorName} denom={props.denom} />
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Validators.js:19]: You should add a key to each list element [ref](https://legacy.reactjs.org/docs/lists-and-keys.html#keys).

File: frontend/src/pages/stakingOverview/Chains.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { Chain } from './Chain'
+
+export const Chains = (props) => {
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Chains.js:4]: This compoment is not required. Move list rendering to parent component.

File: frontend/src/pages/stakingOverview/Chains.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  return (
+    <>
+      {
+        props.chains.map((chain) => <Chain chain={chain} key={chain.chainName} />)
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Chains.js:9]: ditto. Add key to each list element

File: frontend/src/pages/stakingOverview/Validator.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { TableRow, TableCell, Avatar, Box, Typography } from '@mui/material';
+
+export const Validator = (props) => {
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Validator.js:4]: You should add propTypes [ref](https://legacy.reactjs.org/docs/typechecking-with-proptypes.html#proptypes).

File: frontend/src/pages/stakingOverview/Chain.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+          <Grid item>
+            <Button variant="contained" color="primary"
+              disableElevation
+              size="small"
+              sx={{
+                textTransform: "none"
+              }}
+            >
+              Claim:&nbsp;{props.chain.rewards + " " + props.chain.denom}
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Chain.js:42]:
```suggestion
              Claim:&nbsp;{props.chain.rewards}&nbsp;{props.chain.denom}
```

File: frontend/src/pages/stakingOverview/StakingDetails.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { Typography, Grid } from '@mui/material';
+import { Chains } from './Chains';
+
+const StakingDetails = (props) => {
+  return (
+    <Chains chains={props.chains} />
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/StakingDetails.js:7]: ditto here. This component has single element. Move it to parent component.

File: frontend/src/features/staking/stakeSlice.js [hunk 7]
```diff
@@ -307,245 +325,286 @@ export const getUnbonding = createAsyncThunk(
...
   // The `extraReducers` field lets the slice handle actions defined elsewhere,
   // including actions generated by createAsyncThunk or in other slices.
   extraReducers: (builder) => {
     builder
-      .addCase(getValidators.pending, (state) => {
-        state.validators.status = "pending";
-        state.validators.errMsg = "";
+      .addCase(getValidators.pending, (state, action) => {
+        let chainID = action.meta?.arg?.chainID;
+        state.chains[chainID].validators.status = "pending";
```
[the remark this answers is not in the document]
Comment (aleem1314) [reply frontend/src/features/staking/stakeSlice.js:392]: ditto for all other reducers

File: frontend/src/pages/stakingOverview/Validators.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Typography, Table, TableContainer, TableHead, TableBody, TableRow, TableCell } from '@mui/material';
+import { Validator } from './Validator';
+
+export const Validators = (props) => {
+  return (
+    <>
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Validators.js:7]:
```suggestion
```

File: frontend/src/pages/stakingOverview/Validators.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {props.validators.map((validator) => (
+              <Validator validator={validator} key={validator.validatorName} denom={props.denom} />
+            ))}
+          </TableBody>
+        </Table>
+      </TableContainer>
+    </>
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Validators.js:24]:
```suggestion
```

File: frontend/src/pages/stakingOverview/Chains.js (hunk 3 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/Chains.js:9]: This is a bit confusing...
I think I already added key for the list item
<Chain chain={chain} key={chain.chainName}
(the Chain component is the list Item, though will be removing the separate component as suggested)

File: frontend/src/features/staking/stakeSlice.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/features/staking/stakeSlice.js:392]: thats why I added these two reducers..
    we have to call one of these reducers(depeding on the case) when a new chain/chains added to wallet or something..
     Is there any other way to handle this? or should we just give static data?

resetDefaultState : (state, action) => {
      let chainsMap = {};
      let chains = action.payload;
      chains.map ((chainID)=>{chainsMap[chainID]=cloneDeep(initialState.defaultState)});
      state.chains = chainsMap;
    },
    and this
    resetState: (state, action) => {
      let chainID = action.payload.chainID;
      state.chains[chainID].validators = initialState.defaultState.chains[chainID].validators;
      state.chains[chainID].delegations = initialState.defaultState.chains[chainID].delegations;
    },

File: frontend/src/pages/stakingOverview/Validators.js (hunk 1 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/Validators.js:18]: done

File: frontend/src/pages/stakingOverview/Chains.js (hunk 2 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/Chains.js]: done

File: frontend/src/pages/stakingOverview/Validator.js (hunk 4 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/Validator.js:5]: done

File: frontend/src/pages/stakingOverview/StakingDetails.js (hunk 6 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/StakingDetails.js]: done

File: frontend/src/pages/stakingOverview/Validators.js (hunk 8 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/Validators.js]: done

File: frontend/src/pages/stakingOverview/StakingOverview.js [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+        apr: 16.80,
+      }
+    ]
+  }];
+
+  let dummyStakingData = {
+    totalAmount: 13897,
+    totalRewards: 234,
+    chains: chains
+  }
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/StakingOverview.js:143]:
```suggestion
```

File: frontend/src/pages/stakingOverview/StakingOverview.js [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+        <div>
+          <Grid container alignItems="center">
+            <Grid item>
+              <Typography color="text.primary" variant="h4" component="h2">
+                {console.log("here",data)}
+                Staking-{data.chains.length}
+              </Typography>
+            </Grid>
+          </Grid>
+          <div>
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/StakingOverview.js:157]:
```suggestion
```

File: frontend/src/pages/stakingOverview/StakingOverview.js [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+            {data.chains.map((chain) => <Chain chain={chain} key={chain.chainName}/>)}
+          </div>
+        </div>
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/StakingOverview.js:160]:
```suggestion
        </div>
        </>
        :
        null
```

**Comment by aleem1314** [comment]
Need few more changes
- Validator name column is showing validator address.
- Rewards showing `0`.
- Amount is showing in base denom(Should be display denom).
-

**Comment by Teja2045** [comment]
> Need few more changes
> 
>     * Validator name column is showing validator address.
> 
>     * Rewards showing `0`.
> 
>     * Amount is showing in base denom(Should be display denom).

yes anna, I mentioned these issues earlier, I am kind of confused how to resolve those.. https://github.com/vitwit/resolute/pull/484#issuecomment-1576816378

**Comment by aleem1314** [comment]
> * Validator name column is showing validator address.

You can find validator name in `validator.description.moniker`

> * Rewards showing `0`.

Here is an example for rewards calculation https://github.com/vitwit/resolute/blob/master/frontend/src/components/Delegations.jsx#L26-L45

> * Amount is showing in base denom(Should be display denom).

You can get display denom and amount from `coinDecimals` object. To calculate display amount use this formula `Number(baseAmount) * 10 ** currency.coinDecimals` [ref](https://github.com/vitwit/resolute/blob/master/frontend/src/components/Send.js#L45).

File: frontend/src/pages/stakingOverview/StakingTotal.js [hunk 13]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Grid, Card, CardContent, Typography, Button } from '@mui/material';
+
+export const StakingTotal = (props) => {
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/StakingTotal.js:5]: done

File: frontend/src/pages/stakingOverview/StakingOverview.js (hunk 12 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/StakingOverview.js]: done

File: frontend/src/pages/stakingOverview/StakingOverview.js [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+  let dummyStakingData = {
+    totalAmount: 13897,
+    totalRewards: 234,
+    chains: chains
+  }
+  return (
+    <Container>
+      {!(data===undefined || Object.keys(data)===0 || data.chains.length===0) &&  <StakingTotal data={data}/> }
+      {!(data===undefined || Object.keys(data)===0 || data.chains.length===0) &&
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/StakingOverview.js]: done

File: frontend/src/pages/stakingOverview/StakingOverview.js (hunk 11 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/StakingOverview.js]: done

File: frontend/src/pages/stakingOverview/StakingOverview.js (hunk 10 above)
Comment (Teja2045) [reply frontend/src/pages/stakingOverview/StakingOverview.js]: done

File: frontend/src/pages/stakingOverview/Chain.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+        feegranter: feegrant.granter,
+      })
+    );
+  }
+
+  useEffect(() => {
+    dispatch(getDelegatorTotalRewards({chainID:props.chain.chainName, baseURL:chainInfo.network.config.rest, address:chainInfo.walletInfo.bech32Address}));
+  }, [distTxStatus.status])
+
+  console.log(props.chain)
```
Comment (aleem1314) [inline frontend/src/pages/stakingOverview/Chain.js:45]:
```suggestion
```

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**